src/ddmd/dmodule.d

```diff
@@ -112,6 +112,21 @@ extern (C++) const(char)* lookForSourceFile(const(char)** path, const(char)* fil
     return null;
 }
 
+// function used to call semantic3 on a module's dependencies
+void semantic3OnDependencies(Module m)
+{
+    if (!m)
+        return;
+
+    if (m.semanticRun > PASSsemantic3)
+        return;
+
+    m.semantic3(null);
+
+    foreach (i; 1 .. m.aimports.dim)
+        semantic3OnDependencies(m.aimports[i]);
```

Comment:
Okay, but I think we still should figure out why CodeCov says it's uncovered. Knowing whether it's covered in the `-cov` output would tell us whether it's a bug in the test suite or in the CodeCov integration.